### PR TITLE
net-misc/gerbera: Bump to 2.3.0

### PR DESCRIPTION
**Pull request by tietzc**
---

Please check all the boxes that apply:

- [x] I can submit this contribution in agreement with the [Copyright Policy](https://www.gentoo.org/glep/glep-0076.html#certificate-of-origin).
- [x] I have certified the above via adding a `Signed-off-by` line to *every* commit in the pull request.
- [x] This contribution has not been created with the assistance of Natural Language Processing artificial intelligence tools, in accordance with the [AI policy](https://wiki.gentoo.org/wiki/Project:Council/AI_policy).
- [x] I have run `pkgcheck scan --commits --net` to check for issues with my commits.

Please note that all boxes must be checked for the pull request to be merged.

### CODE REVIEW

File: net-misc/gerbera/gerbera-2.3.0.ebuild [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+	dev-libs/spdlog:=
+	net-libs/libupnp:=[ipv6(+),reuseaddr,-blocking-tcp]
+	sys-apps/util-linux
+	sys-libs/zlib
+	virtual/libiconv
+	curl? ( net-misc/curl )
+	exif? ( media-libs/libexif )
+	exiv2? ( media-gfx/exiv2:= )
+	ffmpeg? ( media-video/ffmpeg:= )
+	ffmpegthumbnailer? ( media-video/ffmpegthumbnailer )
```
Comment (juippis) [inline net-misc/gerbera/gerbera-2.3.0.ebuild:39]:
```suggestion
	ffmpegthumbnailer? ( media-video/ffmpegthumbnailer[png] )
```
```
FAILED: gerbera 
: && /usr/bin/x86_64-pc-linux-gnu-g++ -march=native -O2 -pipe -frecord-gcc-switches -Wl,-O1 -Wl,--as-needed -Wl,-z,pack-relative-relocs -Wl,--defsym=__gentoo_check_ldflags__=0 CMakeFiles/gerbera.dir/src/main.cc.o -o gerbera  liblibgerbera.a  -latomic  /usr/lib64/libuuid.so  /usr/lib64/libupnp.so  /usr/lib64/libixml.so  /usr/lib64/libspdlog.so.1.14.1  /usr/lib64/libfmt.so.9.1.0  /usr/lib64/libpugixml.so  /usr/lib64/libsqlite3.so  /usr/lib64/libduktape.so  /usr/lib64/libmagic.so  /usr/lib64/libavformat.so  /usr/lib64/libavutil.so  /usr/lib64/libavcodec.so  /usr/lib64/libffmpegthumbnailer.so  /usr/lib64/libexif.so  /usr/lib64/libz.so && :
/usr/lib/gcc/x86_64-pc-linux-gnu/14/../../../../x86_64-pc-linux-gnu/bin/ld: warning: libpng16.so.16, needed by /usr/lib64/libffmpegthumbnailer.so, not found (try using -rpath or -rpath-link)
/usr/lib/gcc/x86_64-pc-linux-gnu/14/../../../../x86_64-pc-linux-gnu/bin/ld: /usr/lib64/libffmpegthumbnailer.so: undefined reference to `png_set_rows@PNG16_0'
/usr/lib/gcc/x86_64-pc-linux-gnu/14/../../../../x86_64-pc-linux-gnu/bin/ld: /usr/lib64/libffmpegthumbnailer.so: undefined reference to `png_set_IHDR@PNG16_0'
/usr/lib/gcc/x86_64-pc-linux-gnu/14/../../../../x86_64-pc-linux-gnu/bin/ld: /usr/lib64/libffmpegthumbnailer.so: undefined reference to `png_get_io_ptr@PNG16_0'
/usr/lib/gcc/x86_64-pc-linux-gnu/14/../../../../x86_64-pc-linux-gnu/bin/ld: /usr/lib64/libffmpegthumbnailer.so: undefined reference to `png_set_longjmp_fn@PNG16_0'
/usr/lib/gcc/x86_64-pc-linux-gnu/14/../../../../x86_64-pc-linux-gnu/bin/ld: /usr/lib64/libffmpegthumbnailer.so: undefined reference to `png_write_png@PNG16_0'
/usr/lib/gcc/x86_64-pc-linux-gnu/14/../../../../x86_64-pc-linux-gnu/bin/ld: /usr/lib64/libffmpegthumbnailer.so: undefined reference to `png_init_io@PNG16_0'
/usr/lib/gcc/x86_64-pc-linux-gnu/14/../../../../x86_64-pc-linux-gnu/bin/ld: /usr/lib64/libffmpegthumbnailer.so: undefined reference to `png_create_info_struct@PNG16_0'
/usr/lib/gcc/x86_64-pc-linux-gnu/14/../../../../x86_64-pc-linux-gnu/bin/ld: /usr/lib64/libffmpegthumbnailer.so: undefined reference to `png_create_write_struct@PNG16_0'
/usr/lib/gcc/x86_64-pc-linux-gnu/14/../../../../x86_64-pc-linux-gnu/bin/ld: /usr/lib64/libffmpegthumbnailer.so: undefined reference to `png_set_write_fn@PNG16_0'
/usr/lib/gcc/x86_64-pc-linux-gnu/14/../../../../x86_64-pc-linux-gnu/bin/ld: /usr/lib64/libffmpegthumbnailer.so: undefined reference to `png_destroy_write_struct@PNG16_0'
/usr/lib/gcc/x86_64-pc-linux-gnu/14/../../../../x86_64-pc-linux-gnu/bin/ld: /usr/lib64/libffmpegthumbnailer.so: undefined reference to `png_set_text@PNG16_0'
collect2: error: ld returned 1 exit status
ninja: build stopped: subcommand failed.
 * ERROR: net-misc/gerbera-2.3.0::gentoo failed (compile phase):
```

File: net-misc/gerbera/gerbera-2.3.0.ebuild [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+src_install() {
+	cmake_src_install
+
+	newinitd "${FILESDIR}"/${PN}-1.0.0.initd ${PN}
+	newconfd "${FILESDIR}"/${PN}-1.0.0.confd ${PN}
+
+	insinto /etc/${PN}
+	newins "${FILESDIR}"/${PN}-1.3.0.config config.xml
+	fperms 0640 /etc/${PN}/config.xml
+	fowners root:gerbera /etc/${PN}/config.xml
```
Comment (juippis) [inline net-misc/gerbera/gerbera-2.3.0.ebuild:81]: Looks like you need to depend on the group, not the user.

**Review by juippis — COMMENTED** [review]
I see 2.3.0 added a new build option, 
`option(BUILD_DOC              "Add 'doc' target to generate documentation" OFF)`
could you add it and see what dependencies it possibly has to get it working?

**Comment by tietzc** [comment]
Suggestions from @juippis have been implemented.

**Review by juippis — COMMENTED** [review]
Sorry to bug you, but I don't see any difference in installed files whether `doc` is enabled or disabled. You may need a separate section in `src_install` to actually install the generated doc files after they're built with the use flag?

**Comment by tietzc** [comment]
@juippis In fact it looks like the documentation is not being built at all. If USE doc is enabled the build system successfully checks for doxygen and graphviz, but then nothing further seems to happen in regards to the documentation. I am sorry, but I do not know how to fix this. Any help would be greatly appreciated.

**Comment by juippis** [comment]
We most likely have to enter the doc directory manually within the ebuild, and call **doxygen .** there. I'll take a look later today, and move the deps to BDEPEND as well. We have an eclass to help here, https://devmanual.gentoo.org/eclass-reference/docs.eclass/index.html but I'm pretty confident it's a simple fix I can do quickly some point today.

**Comment by tietzc** [comment]
@juippis Thank you very much! Just one question: Shouldn’t BDEPEND be put behind the doc USE flag so that it doesn’t pull in sphinx and graphviz unconditionally?

**Comment by juippis** [comment]
Oh, right, yeah I couldn't get `DOCS_DEPEND` to work nicely with the deps so I hardcoded the deps **for testing only**, but since it worked didn't take a look at the ebuild again :( will fix, thanks for notifying me!

**Comment by tietzc** [comment]
@juippis Thanks! Sorry to bother you again, but it appears like you forgot to add the doxygen dependency, which is required by the build system if USE doc is enabled. See https://bugs.gentoo.org/941911

Could you please add app-text/doxygen to BDEPEND behind the doc USE flag? Thank you in advance.

**Comment by juippis** [comment]
5845b372a763c9e3b1dee4e8c118bb04171cad7f that's an upstream bug though.

**Comment by tietzc** [comment]
@juippis Thanks! Now we should be good! I was surprised as well, as it actually uses sphinx to build the docs. Might be worth poking upstream. Anyway, thanks a lot for your help here.

**Pull request closed without merging**